src/mndragossh/week3/shop/Basket.java

```diff
@@ -0,0 +1,34 @@
+package src.mndragossh.week3.shop;
+
+import java.util.ArrayList;
+
+public class Basket {
+    Customer customer;
+    ArrayList<Product> products;
+
+    public Basket(Customer customer) {
+        this.customer = customer;
+        products = new ArrayList<>();
+    }
+
+    public void addToBasket(Product product) {
+        products.add(product);
+    }
+
+    public void removeFromBasket(Product product) {
+        products.remove(product);
+    }
+
+    public double getTotalPrice() {
+       if(customer.membership != null) {
+           if (customer.membership.equals(Membership.Gold)) {
+               return products.stream().mapToDouble(x -> x.price).sum() - products.stream().mapToDouble(x -> x.price).sum() * 0.3;

```

Comment:
fain ca te-ai folosit de stream; ai putea sa imbunatatesti bucata asta daca ai refactoriza putin si ai calcula o singura data: `double total = products.stream().mapToDouble(x -> x.price).sum();`
iar pentru 0.3 valoarea discountului ar fi mai bine sa adaugi in enum un field care sa pasteze discountul pentru ca daca o sa vrei sa te folosesti de membership prin mai multe locuri si il scrii pe peste tot 0.3, daca te hotarasti la un moment dat sa il schimbi o sa fie nevoie sa cauti toate locurile in care l-ai hardcodat si sa faci modificarea; daca il tii ca field in enum, o sa faci schimbarea intr-un singur loc